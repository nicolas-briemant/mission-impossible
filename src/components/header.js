import React from 'react';
import PropTypes from 'prop-types';
import { Navbar, NavbarGroup, Classes } from "@blueprintjs/core";
import cx from 'classnames';
import glamorous from 'glamorous';

export const HeaderLeft = ({ children }) => <NavbarGroup aligh="left">{children}</NavbarGroup>;
HeaderLeft.propTypes = { children: PropTypes.node };

export const HeaderRight = ({ children }) => <NavbarGroup align="right">{children}</NavbarGroup>;
HeaderRight.propTypes = { children: PropTypes.node };

const StyledHeader = glamorous(Navbar)(
  {},
  ({ isfixed }) => ({ zIndex: isfixed === 'true' ? 10 : 0 })
);

export const Header = ({ isFixed, children }) => {
  const headerLeft = () => React.Children.toArray(children).find((child) => child.type === HeaderLeft);
  const headerRight = () => React.Children.toArray(children).find(({type}) => type === HeaderRight);

  return (
    <StyledHeader
      isfixed={isFixed ? 'true' : 'false'}
      className={cx({[Classes.FIXED_TOP]: isFixed})}
    >
      {headerLeft()}
      {headerRight()}
    </StyledHeader>
  );
};

Header.propTypes = {
  isFixed: PropTypes.bool,
  children: PropTypes.node,
};
