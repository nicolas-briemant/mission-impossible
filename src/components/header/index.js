import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';

export const HeaderLeft = ({ children }) => <Fragment>{children}</Fragment>;

HeaderLeft.propTypes = {
  children: PropTypes.node,
};

export const HeaderRight = ({ children }) => <Fragment>{children}</Fragment>;

HeaderRight.propTypes = {
  children: PropTypes.node,
};

const HeaderContain = glamorous.div({
  position: 'relative',
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
});

export const Header = ({ children }) => {
  const headerLeft = () => React.Children.toArray(children).find(({ type }) => type === HeaderLeft);
  const headerRight = () => React.Children.toArray(children).find(({ type }) => type === HeaderRight);
  return (
    <HeaderContain>
      <div>{headerLeft()}</div>
      <div>{headerRight()}</div>
    </HeaderContain>
  );
};

Header.propTypes = {
  children: PropTypes.node,
};
