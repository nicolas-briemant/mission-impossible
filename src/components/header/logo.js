import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { css } from 'glamor';

const HeaderContainLogoAndTitle = glamorous.div({
  display: 'flex',
  alignItems: 'center',
});

const anim = css.keyframes({
  '0%': { transform: 'rotateY(0deg)' },
  '25%': { transform: 'rotateY(90deg)' },
  '50%': { transform: 'rotateY(180deg)' },
  '75%': { transform: 'rotateY(270deg)' },
  '100%': { transform: 'rotateY(360deg)' },
});

const HeaderLogoImg = glamorous.img({
  height: '77px',
  width: 'auto',
  animation: `${anim} 5s infinite linear`,
});

const HeaderContainTitle = glamorous.div({
  position: 'relative',
  justifyContent: 'center',
});

const HeaderTitle = glamorous.h1({
  fontSize: '20px',
});

const Logo = ({ src, alt, title }) => {
  return (
    <HeaderContainLogoAndTitle>
      <HeaderLogoImg src={src} alt={alt} />
      <HeaderContainTitle>
        <HeaderTitle>{title}</HeaderTitle>
      </HeaderContainTitle>
    </HeaderContainLogoAndTitle>
  );
};

Logo.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default Logo;
