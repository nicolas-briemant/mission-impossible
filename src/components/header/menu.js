import React from 'react';
import glamorous from 'glamorous';
import cx from 'classnames';
import PropTypes from 'prop-types';
import { Menu, MenuItem, MenuDivider, Classes } from '@blueprintjs/core';

const HeaderContainMenu = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
});

const HeaderMenu = ({ blockedAction }) => {
  return (
    <HeaderContainMenu>
      <Menu>
        <MenuItem className={cx(Classes.BUTTON, Classes.ACTIVE)} iconName="link" text="Mission" />
        <MenuItem
          className={cx(Classes.BUTTON)}
          iconName="link"
          text="Shall Not Pass"
          onClick={() => blockedAction('Shall Not Pass')}
        />
        <MenuDivider />
      </Menu>
    </HeaderContainMenu>
  );
};

HeaderMenu.propTypes = {
  blockedAction: PropTypes.func.isRequired,
};

export default HeaderMenu;
