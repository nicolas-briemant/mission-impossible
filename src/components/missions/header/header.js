import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { Navbar, NavbarGroup } from '@blueprintjs/core';
import 'normalize.css/normalize.css';

const MyNavbarGroup = glamorous(NavbarGroup)({
  display: 'flex',
  alignItems: 'center',
});

export const HeaderLeft = ({ children }) => <Fragment>{children}</Fragment>;

HeaderLeft.propTypes = {
  children: PropTypes.node,
};

export const HeaderRight = ({ children }) => <Fragment>{children}</Fragment>;

HeaderRight.propTypes = {
  children: PropTypes.node,
};

export const Header = ({ children }) => {
  const headerLeft = () => React.Children.toArray(children).find(({ type }) => type === HeaderLeft);
  const headerRight = () => React.Children.toArray(children).find(({ type }) => type === HeaderRight);

  return (
    <header>
      <Navbar>
        <MyNavbarGroup align="left">{headerLeft()}</MyNavbarGroup>
        <MyNavbarGroup align="right">{headerRight()}</MyNavbarGroup>
      </Navbar>
    </header>
  );
};

Header.propTypes = {
  children: PropTypes.node,
};
