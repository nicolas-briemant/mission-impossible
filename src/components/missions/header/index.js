import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { ButtonGroup, Button } from '@blueprintjs/core';
import { Header, HeaderLeft, HeaderRight } from './header';
import logo from '../../../data/logo.png';

const Logo = glamorous.img({
  width: 75,
});

const Header2 = ({ blockAction }) => (
  <Header>
    <HeaderLeft>
      <Logo alt="Logo" src={logo} />
      <h4>Titre du site</h4>
    </HeaderLeft>
    <HeaderRight>
      <ButtonGroup minimal="true" large="true">
        <Button iconName="pt-icon-align-justify" onClick={() => blockAction()}>
          Missions
        </Button>
        <Button iconName="people" onClick={() => blockAction()}>
          Peoples
        </Button>
      </ButtonGroup>
    </HeaderRight>
  </Header>
);

Header2.propTypes = {
  blockAction: PropTypes.func,
};

export default Header2;
