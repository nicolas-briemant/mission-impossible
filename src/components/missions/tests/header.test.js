import React from 'react';
import { shallow, mount } from 'enzyme';
import { HeaderFull as Header, HeaderLeft, HeaderRight } from '../header/header';

describe('app | component | header', () => {
  describe('render', () => {
    test('should match snapshot', () => {
      const wrapper = shallow(<Header />);
      expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with children left', () => {
      const root = (
        <Header>
          <HeaderLeft>
            <span>Left</span>
          </HeaderLeft>
          <HeaderRight>
            <span>Right</span>
          </HeaderRight>
        </Header>
      );
      const wrapper = mount(root);
      expect(wrapper.find(HeaderLeft)).toHaveLength(1);
    });
    test('should match snapshot with children right', () => {
      const root = (
        <Header>
          <HeaderLeft>
            <span>Left</span>
          </HeaderLeft>
          <HeaderRight>
            <span>Right</span>
          </HeaderRight>
        </Header>
      );
      const wrapper = mount(root);
      expect(wrapper.find(HeaderRight)).toHaveLength(1);
    });
    test('should match snapshot with children left content', () => {
      const root = (
        <Header>
          <HeaderLeft>
            <span>Left</span>
          </HeaderLeft>
          <HeaderRight>
            <span>Right</span>
          </HeaderRight>
        </Header>
      );
      const wrapper = mount(root);
      expect(wrapper.find(HeaderLeft).contains(<span>Left</span>)).toEqual(true);
    });
    test('should match snapshot with children right content', () => {
      const root = (
        <Header>
          <HeaderLeft>
            <span>Left</span>
          </HeaderLeft>
          <HeaderRight>
            <span>Right</span>
          </HeaderRight>
        </Header>
      );
      const wrapper = mount(root);
      expect(wrapper.find(HeaderRight).contains(<span>Right</span>)).toEqual(true);
    });
  });
});
