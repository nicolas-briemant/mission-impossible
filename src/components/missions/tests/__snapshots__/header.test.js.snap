// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app | component | header render should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Blueprint.Navbar>
        <Blueprint.NavbarGroup
                align="left"
                className={
                        Object {
                                "data-css-12alag6": "",
                              }
                }
        />
        <Blueprint.NavbarGroup
                align="right"
                className={
                        Object {
                                "data-css-12alag6": "",
                              }
                }
        />
</Blueprint.Navbar>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Blueprint.NavbarGroup
            align="left"
            className={
                        Object {
                                    "data-css-12alag6": "",
                                  }
            }
/>,
          <Blueprint.NavbarGroup
            align="right"
            className={
                        Object {
                                    "data-css-12alag6": "",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "left",
            "children": undefined,
            "className": Object {
              "data-css-12alag6": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "right",
            "children": undefined,
            "className": Object {
              "data-css-12alag6": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Blueprint.Navbar>
          <Blueprint.NavbarGroup
                    align="left"
                    className={
                              Object {
                                        "data-css-12alag6": "",
                                      }
                    }
          />
          <Blueprint.NavbarGroup
                    align="right"
                    className={
                              Object {
                                        "data-css-12alag6": "",
                                      }
                    }
          />
</Blueprint.Navbar>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Blueprint.NavbarGroup
              align="left"
              className={
                            Object {
                                          "data-css-12alag6": "",
                                        }
              }
/>,
            <Blueprint.NavbarGroup
              align="right"
              className={
                            Object {
                                          "data-css-12alag6": "",
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "left",
              "children": undefined,
              "className": Object {
                "data-css-12alag6": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "right",
              "children": undefined,
              "className": Object {
                "data-css-12alag6": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
